src/App.tsx

```
@@ -1,26 +1,256 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { TodoForm } from './components/TodoForm';
+import { TodoList } from './components/TodoList';
+import { Notification } from './components/Notification';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 3135;
+const API_URL = 'https://mate.academy/students-api/todos';
+
+type Filter = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [title, setTitle] = useState('');
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const [error, setError] = useState('');
+  const [filter, setFilter] = useState<Filter>('all');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const showError = (msg: string) => {
+    setError(msg);
+    setTimeout(() => setError(''), 3000);
+  };
+
+  useEffect(() => {
+    fetch(`${API_URL}?userId=${USER_ID}`)
+      .then(res => {
+        if (!res.ok) {
+          throw new Error();
+        }
+
+        return res.json();
+      })
+      .then(setTodos)
+      .catch(() => showError('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (tempTodo === null) {
+      inputRef.current?.focus();
+    }
+  }, [tempTodo]);
+
+  const handleAdd = async (e: React.FormEvent) => {
+    e.preventDefault();
+    const trimmed = title.trim();
+
+    if (!trimmed) {
+      showError('Title should not be empty');
```

Instead of passing arbitrary values into showError define fixed set of possible error types to ensure consistency and maintainability